core/src/main/java/edu/wpi/grip/core/sources/HttpSource.java

```diff
@@ -0,0 +1,102 @@
+
+package edu.wpi.grip.core.sources;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Inject;
+
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.SocketHint;
+import edu.wpi.grip.core.SocketHints;
+import edu.wpi.grip.core.Source;
+import edu.wpi.grip.core.events.SourceHasPendingUpdateEvent;
+import edu.wpi.grip.core.events.SourceRemovedEvent;
+import edu.wpi.grip.core.http.GripServer;
+import edu.wpi.grip.core.http.PostHandler;
+import edu.wpi.grip.core.util.ExceptionWitness;
+
+import java.util.Properties;
+
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacpp.opencv_imgcodecs;
+
+/**
+ * Provides a way to generate a constantly updated {@link Mat} from the internal HTTP server.
+ */
+public class HttpSource extends Source implements PostHandler {
+
+    private final OutputSocket<Mat> imageOutput;
+    private final SocketHint<Mat> outputHint = SocketHints.Outputs.createMatSocketHint("Image");
+    private byte[] data = new byte[0];
+    private boolean gotImage = false;
+
+    private final EventBus eventBus;
+    private final GripServer server;
+
+    public interface Factory {
+        HttpSource create();
+    }
+
+    @Inject
+    HttpSource(
+            ExceptionWitness.Factory exceptionWitnessFactory,
+            EventBus eventBus,
+            GripServer server) {
+        super(exceptionWitnessFactory);
+        this.eventBus = eventBus;
+        this.server = server;
+        this.imageOutput = new OutputSocket<>(eventBus, outputHint);
+    }
+
+    @Override
+    public String getName() {
+        return "HTTP source";
+    }
+
+    @Override
+    protected OutputSocket[] createOutputSockets() {
+        return new OutputSocket[]{imageOutput};
+    }
+
+    @Override
+    protected boolean updateOutputSockets() {
+        if (!gotImage) {
+            return false;
+        }
+        if (data.length == 0) {
+            // Got data, but it's empty
+            return false;
+        }
+        gotImage = false;
+        imageOutput.setValue(opencv_imgcodecs.imdecode(new Mat(data), opencv_imgcodecs.CV_LOAD_IMAGE_COLOR));
```

Comment:
This could cause a potential memory leak.
You are allocating a new mat for every image uploaded.
What is the fastest rate you could see images moving through this source?